### PR TITLE
Fix parsing SupportedCtapOptions from empty JSON object

### PR DESCRIPTION
**Pull request by emlun**
(This would merge into #419, not into main)

Out of curiosity I tried what happens if you add a [test of parsing from an empty JSON object](https://github.com/Yubico/java-webauthn-server/pull/419#discussion_r2123762009), and it turned out to have wider repercussions than I expected :smile: it also interacts with the builder annotations in subtle ways. I've documented the process in detail in the first two commit descriptions. The last two are just cleanup and gathering related things closer together in the source code.

### CODE REVIEW

File: webauthn-server-attestation/src/main/java/com/yubico/fido/metadata/SupportedCtapOptions.java [hunk 1]
```diff
@@ -17,145 +17,147 @@
  */
 @Value
 @Builder
```
Comment (fdennis) [inline webauthn-server-attestation/src/main/java/com/yubico/fido/metadata/SupportedCtapOptions.java:19]: Is this still needed?

**Review by fdennis — APPROVED** [review]
